### ISSUE TITLE
Add equivalent cURL --resolve flag

### ISSUE THREAD

**Issue by aventurella** [jkbrzt/httpie#99]
Per cURL --help:

--resolve HOST:PORT:ADDRESS  Force resolve of HOST:PORT to ADDRESS

(bypasses hosts file)

example usage:

`http --resolve "www.foo.com:127.0.0.1" GET www.foo.com/whatever`

OR

`http --resolve "www.foo.com:8080:127.0.0.1" GET www.foo.com:8080/whatever`

**Comment by CMCDragonkai** [comment]
The above trick doesn't work for HTTPS sites. We need a full resolve option similar to curl.

**Comment by MadMub** [comment]
I'm using httpie to test that I setup my DNS round-robin correctly. +1 for feature

**Comment by aztlan2k** [comment]
Just curious...  This has been open for over a year now (since it was reopened)... any chance it's getting looked at?  Seems like a useful feature to have.

**Comment by sigmavirus24** [comment]
The underlying software that HTTPie uses does not allow people to specify their own IP addresses for a specific host. It's unlikely it will be added there. So, @aztlan2k it's not likely this is being worked on. Sorry. :/

**Comment by dair-targ** [comment]
Hi! I've submitted some code and would appreciate any comments.

**Comment by jakubroztocil** [comment]
@dair-targ please feel free to open a pull request for easier discussion on the code

**Comment by dair-targ** [comment]
Just re-submitted via another PR. Would you mind to check?

**Comment by pcgeek86** [comment]
Any chance of getting this merged? @jakubroztocil

**Comment by gregfr** [comment]
That would be great to prepare site migration and avoid problems!

**Comment by dimisjim** [comment]
This can be achieved via `dig +short <host>`. It's a DNS thing, not really HTTP

**Comment by lapo-luchini** [comment]
`dig` is used to query the DNS, the proposed `--resove` is used to force resolution of an address during an HTTP fetch (probably to a value the DNS server doesn't yet have but will have in the future, after this testing is done).
It's rather an alternative to editing `/etc/hosts` than it is to using `dig` (or `drill`, or similar).

**Comment by tn-osimis** [comment]
In case anybody is confused as to why this is needed "for HTTPS", it's specifically when TLS [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) (Server Name Indication) is used in place of the HTTP Host header to disambiguate connections (e.g. hostname-based L7 routing in reverse-proxies, or "virtual server" use-cases in general), which is extremely common.

In that context, resolving the IP beforehand isn't sufficient as it won't constitute a virtual server name expected by the actual server.

If an equivalent to `--resolve` isn't desirable for any reason, forcing the SNI value (e.g. `--sni`) would be an acceptable alternative IMO since we can already do the equivalent for plain HTTP (i.e. set the Host header value explicitly).

**Comment by segevfiner** [comment]
This can also be useful for absolute redirects (That include a full URL with a host name)

**Comment by Almad** [comment]
Looking into it, there are basically two options how to fix this:

* Work with requests/urllib3 to add support to the lower-level libraries. There is a PR for this at https://github.com/urllib3/urllib3/pull/1209 , which got abandoned during their "stale PR purge". Picking it up and working with urllib3 to add it is an option
* Monkeypatch sockets. This was proposed in #659 

The strong preference would be for the first solution. Custom wrapping and  monkeypatching often leads to unintended consequences as the Python interpreter and surrounding libraries develop and often leads to "hidden" bugs.

Help with getting the patch rolling for the urllib3 would be very welcome! Otherwise, we may look into it in the future, but other features have higher priority.

**Comment by hussam-qasem** [comment]
Greetings! I just realized this thread is over 10 years old.
Is the status (blocked by upstream) still current?
Thank you!!

**Comment by AndrewKvalheim** [comment]
The Rust reimplementation [xh](https://github.com/ducaale/xh) has added `--resolve`.
